### ISSUE TITLE
Preamble not refreshing on change of focus

### ISSUE THREAD

**Issue by abbas-sabra-sonarsource** [SonarSource/sonarlint-visualstudio#2242]
To reproduce: change the focus from any C++ file to the output windows. Then change the focus from the output window back to the file.
  
Tested working version: 4.28
The preamble is refreshed. The message is printed in the SonarLint output windows.
Output: 
![image](https://user-images.githubusercontent.com/49131500/112024419-760c2600-8b34-11eb-9a76-6653abaa0d53.png)

Tested buggy version 4.34
Doing the same steps doesn't lead to any message in the SonarLint output windows. The preamble is no refreshed.

**Comment by duncanp-sonar** [comment]
@abbas-sabra-sonarsource does this change of behaviour cause a problem?

We fixed a couple of focus-related bugs in the last version (#2079 and #2091) so that we now only react to changes from one document window to another. So if you are on document window A, then switch to a tool window, then switch back to A, we won't refresh the preamble, but we shouldn't need to because the last-active document hasn't changed.

If you switch from doc window A to a tool window to doc window B, we'd expect the preamble to be refreshed.

**Comment by abbas-sabra-sonarsource** [comment]
@duncanp-sonar yes it does cause a problem. The preamble is not only dependent on the active document. You don't have to change the file to have an impact on it. You can simply change your build option from debug to release(the preamble should be refreshed in this case). Or you might go to the setting and change the compilation arguments which impact the preamble. There are also other special scenarios. For example, I sometimes edit some outside headers that end up in the preamble from VScode. 
In short, there are many things that impact the preamble, and refreshing it on focus was done on purpose to make sure that you will never run the analysis with a bad preamble.

**Comment by duncanp-sonar** [comment]
Thanks for the additional info @abbas-sabra-sonarsource
